### ISSUE TITLE
301 not triggered

### ISSUE THREAD

**Issue by loudapet** [loudapet/42webserv#67]
301 does not trigger when receiving a folder without a slash as the target resource. That happens because the matched location path is replaced by root, which includes slash, and the check proceeds to check the final path and throws a 403 when autoindexing is not allowed, instead of throwing a 301 and sending Content-Location with the proper way to handle it.

Example:
```bash
path:	/tours             #path as parsed from config file
root:	docs/our_web/      #root of the location as per config file
URL:	/tours             #incoming URI from the request
Final path: docs/our_web/  #part of the URL matching path replaced by root
```

**Comment by loudapet** [comment]
> Our config:

```
location /tours {
    root docs/our_web;
    ...
}
```

> Nginx config:

```
location /lmao {
    root   docs/our_web;
    ...
}
```

The issue is that a slash appends to `root` on both cases, but since we're treating rooting as `alias` directive from nginx, and therefore replace the location path with `root`, we end up with the following

> `docs/our_web/`

instead of nginx's

> `docs/our_web/lmao`

which in turn prevents 301 from ever triggering (the folder will always end with a slash, so we will never manage to target a folder in file form).

tl;dr it's a feature, not a bug!
@andreaulicna pls confirm that I'm not just seeing things and that I can actually close this with a valid explanation that makes sense to you.

**Comment by andreaulicna** [comment]
The explanation in the previous comment very much makes sense, it's what the subject is asking us to do with the location matching after all! so I'm closing this :)
